crypto/hash.go

```diff
@@ -2,76 +2,70 @@ package crypto
 
 import (
 	"encoding/hex"
+
 	"golang.org/x/crypto/blake2b"
 	"golang.org/x/crypto/ripemd160"
 )
 
-func Blake256(data string) string {
+func Blake256(data []byte) []byte {
 	b, _ := blake2b.New(32, nil)
-	s, _ := hex.DecodeString(data)
-	b.Write(s)
+	b.Write(data)
 	bs := b.Sum(nil)
 	var result = make([]byte, 64)
 	hex.Encode(result, bs[:])
-	return string(result)
+	return result
 }
 
-func Blake256WithKey(data string, key []byte) string {
+func Blake256WithKey(data []byte, key []byte) []byte {
 	b, _ := blake2b.New(32, key)
-	s, _ := hex.DecodeString(data)
-	b.Write(s)
+	b.Write(data)
 	bs := b.Sum(nil)
 	var result = make([]byte, 64)
 	hex.Encode(result, bs[:])
-	return string(result)
+	return result
 }
 
-func Blake128(data string) string {
+func Blake128(data []byte) []byte {
 	b, _ := blake2b.New(16, nil)
-	s, _ := hex.DecodeString(data)
-	b.Write(s)
+	b.Write(data)
 	bs := b.Sum(nil)
 	var result = make([]byte, 32)
 	hex.Encode(result, bs[:])
-	return string(result)
+	return result
```

Comment:
Let's return binary data directly. Returning hex-encoded value in []byte type is strange.